error.go

```diff
@@ -0,0 +1,10 @@
+package purego
+
+// Error represents an error value returned from purego
+type Error struct {
```

Comment:
@TotallyGamerJet Don't we need to distinguish libdl errors or others? If so, we might want a special type for libdl error

Reply:
I added this error type specifically for libdl errors, so I it should already be distinguished from other errors. I can make the name more specific, like `LibdlError`, if you think that would be better.

Reply:
`DlError` sounds better (if we decide to remove the `Dlerror` function)

Reply:
I think Dlerror sounds good